### PR TITLE
Bug 1992730: Fix i18n for dynamic plugins

### PR DESCRIPTION
**Pull request by spadgett**
* Add plugin namespaces when initializing i18next
* Add react-i18next to the shared modules
* Update the demo plugin to use translations

/cc @christianvogt @bipuladh @rebeccaalpert @jhadvig

### CODE REVIEW

**Comment by spadgett** [comment]
This works running a dev environment, but deploying a plugin to a real cluster results in

> react-i18next:: You will need to pass in an i18next instance by using initReactI18next

**Comment by spadgett** [comment]
> This works running a dev environment, but deploying a plugin to a real cluster results in
> 
> > react-i18next:: You will need to pass in an i18next instance by using initReactI18next

Adding even a single nav item to the plugin makes this warning go away (even if I still navigate directly to the route with the problem). I suspect there is a race where the page is rendered before i18next is initialized. @christianvogt @vojtechszocs Any ideas about this?

I might open a separate bug to investigate the race since this PR fixes the basic problem of i18n not working at all.

File: frontend/dynamic-demo-plugin/README.md [hunk 1]
```diff
@@ -83,3 +83,22 @@ Following commands should be executed in Console repository root.
...
+with this namespace as follows:
+
+```tsx
+conster Header: React.FC = () => {
+  const { t } = useTranslation('plugin__console-demo-plugin);
+  return <h1>{t('Hello, World!')}</h1>;
+};
+```
+
+Running `yarn i18n` updates the JSON files in the locales folder of the dynamic
```
Comment (vojtechszocs) [inline frontend/dynamic-demo-plugin/README.md:103]:
```suggestion
Running `yarn i18n` updates the JSON files in the `locales` folder of the dynamic
```

File: frontend/dynamic-demo-plugin/i18next-parser.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const FALLBACK_LOCALE = 'en';
```
Comment (vojtechszocs) [inline frontend/dynamic-demo-plugin/i18next-parser.config.js:1]: Is this actually needed?

I can't see this option listed in [`i18next-parser` options](https://github.com/i18next/i18next-parser#options) :thinking:

**Review by vojtechszocs — CHANGES_REQUESTED** [review]

**Comment by vojtechszocs** [comment]
> Adding even a single nav item to the plugin makes this warning go away (even if I still navigate directly to the route with the problem). I suspect there is a race where the page is rendered before i18next is initialized. @christianvogt @vojtechszocs Any ideas about this?
> 
> I might open a separate bug to investigate the race since this PR fixes the basic problem of i18n not working at all.

`i18next` configuration & initialization seems to be coupled together in `public/i18n.js` and this module gets imported (for programmatic access to `t` function) from lots of other modules.

@spadgett Maybe we should import `public/i18n.js` inside `public/components/app.jsx` to ensure early evaluation of its logic.

(Ideally, config vs. init code should be separated but this can be influenced by `i18next` usage pattern.)

**Comment by spadgett** [comment]
Thanks, @vojtechszocs. I'm a little confused about the imports. To use `t` you should be able to do

```ts
import i18next from 'i18next';
// ...
i18next.t('pipelines-plugin~Name');
// etc
```

In the original PR, `i18n.js` was only imported from console-shared:

https://github.com/openshift/console/commit/5464e1058c275f95a29688fbe2f5c192de0f78b7#diff-d52a3952d7f2b78f48ab59ce7b1afc3ca2ff992b9b383012f5b79e3018525274R1

I don't know if those other imports were added to fix something or if they were just copy/pasted. I expect we'd only want to import it from one place, probably app.jsx.

cc @rebeccaalpert

File: frontend/dynamic-demo-plugin/i18next-parser.config.js (hunk 2 above)
Comment (spadgett) [reply frontend/dynamic-demo-plugin/i18next-parser.config.js:1]: This was copied and pasted from here:

https://github.com/spadgett/console/blob/093d3fee0aae7d4af86936fa64f43acfca556d61/frontend/i18next-parser.config.js#L3

But I don't see where it is ever used. That was probably meant to be `fallbackLng`. I'll take it out.

cc @rebeccaalpert

**Comment by spadgett** [comment]
Pushed updates for the review comments.

**Pull request merged**